deps/ncrypto/ncrypto.cc

```diff
@@ -220,4 +221,72 @@ int PasswordCallback(char* buf, int size, int rwflag, void* u) {
   return -1;
 }
 
+// ============================================================================
+// SPKAC
+
+bool VerifySpkac(const char* input, size_t length) {
+#ifdef OPENSSL_IS_BORINGSSL
+  // OpenSSL uses EVP_DecodeBlock, which explicitly removes trailing characters,
+  // while BoringSSL uses EVP_DecodedLength and EVP_DecodeBase64, which do not.
+  // As such, we trim those characters here for compatibility.
+  //
+  // find_last_not_of can return npos, which is the maximum value of size_t.
+  // The + 1 will force a roll-ver to 0, which is the correct value. in that
+  // case.
+  length = std::string_view(input, length).find_last_not_of(" \n\r\t") + 1;
```

Comment:
We need something like this, in all places

```suggestion
  auto idx = std::string_view(input, length).find_last_not_of(" \n\r\t");
  CHECK_NE(idx, std::string_view::npos);
  length = idx + 1;
```

Reply:
It's perfectly acceptable for `idx` here to be returned as `npos` tho. The `+ 1` rolls it over to ensure that `length` is always a valid value. Otherwise, if someone passed in literally something like `" \n\r\t"` as the value, which should be interpreted as zero-length, this code path would end up throwing an assertion.